Data.Rmd

```diff
@@ -0,0 +1,567 @@
+-   usethis: Most functions need to know the active project, i.e. which directory to target for file modification. This is often the current working directory, but that is not an invariant usethis can rely upon. One potential design is to make it possible to specify the target project as an argument of every function in usethis. But this would create significant clutter in the user interface, as well as internal fussiness. Instead, we determine the active project upon first need, cache it, and provide methods for (re)setting it.
+
+The blog post [Package-Wide Variables/Cache in R Packages](https://trestletech.com/2013/04/package-wide-variablescache-in-r-package/) gives a more detailed development of this technique.
+
+<!-- I've always felt like Hermione's beaded bag / Bag of Holding is a great analogy for this environment technique. As long as you've got this bag, you can keep whatever you like in it. Should I try to develop this? -->
+
+## Persistent user data
+
+Sometimes there is data that your package obtains, on behalf of itself or the user, that should persist *even across R sessions*.
+This is our last and probably least common form of storing package data.
+For the data to persist this way, it has to be stored on disk and the big question is where to write such a file.
+
+This problem is hardly unique to R.
+Many applications need to leave notes to themselves.
+It is best to comply with external conventions, which in this case means the [XDG Base Directory Specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html).
+You need to use the official locations for persistent file storage, because it's the responsible and courteous thing to do and also to comply with CRAN policies.
+In particular, you can't just write into the user's home directory.
+Here's a relevant excerpt from the CRAN policy at the time of writing:
+
+> Packages should not write in the user's home filespace (including clipboards), nor anywhere else on the file system apart from the R session's temporary directory ....
+>
+> For R version 4.0 or later (hence a version dependency is required or only conditional use is possible), packages may store user-specific data, configuration and cache files in their respective user directories obtained from `tools::R_user_dir()`, provided that by \[sic\] default sizes are kept as small as possible and the contents are actively managed (including removing outdated material).
+
+In R, there are two tools you can use to derive acceptable locations for user data:
+
+-   The `tools::R_user_dir()` function, available in R \>= 4.0.
+-   The [rappdirs package](https://rappdirs.r-lib.org), which is a port of the Python appdirs module, compatible with R \>= 3.5[^data-3].
```

Comment:
I think you can drop rappdirs, or just mention it in a footnote.

Reply:
Moved to footnote